### ISSUE TITLE
How to run this angular 4 app standalone?

### ISSUE THREAD

**Issue by shobhitshukla** [me-12/single-spa-portal-example#18]
First of all, a great thanks for this example!

As in this example portal is working fine with all child apps but I need to run angular app standalone as well.

Will you please provide me the solution to run angular app standalone as you did for Vue project.

Thanks again!

**Comment by naumannt** [comment]
The .README says that the watch:standalone script is exclusive to the vue-project. Like the OP of this thread I would like to run an Angular-App on its own as well, but I'm not quite sure how I could achieve that through a webpack-task (since the watch:portal already basically just runs the application). Would a task like that need to modify configuration/code first?

**Comment by ssi-hu-tasi-norbert** [comment]
Simply add a main.ts to the root and use ng serve as usual. The index.html and main.ts will be used only the cli, the portal uses the main-module.ts. Due to zone.js is commented out in polyfills.ts you have to add this into body in index.html:
`<script src="https://cdnjs.cloudflare.com/ajax/libs/zone.js/0.8.19/zone.js"></script> `

**Comment by idzer0lis** [comment]
@ssi-hu-tasi-norbert I believe there is a a need for angular.json too. Could you explain a little better how did you achieved running angular micro app using the cli? I got blank page and no errors

**Comment by ssi-hu-tasi-norbert** [comment]
Yes, you need everything that cli generates. What you have in main-module(providers, import, etc..) you have to insert into app.module.ts (redundantly) too. The cli generates main.ts, app.module.ts, index.html that are not used by the portal. Of course you cannot access the globaleventdispatcher.

**Comment by idzer0lis** [comment]
@ssi-hu-tasi-norbert thanks! Did you manage to also build the angular app this way?

**Comment by ssi-hu-tasi-norbert** [comment]
You're welcome!
I could manage to build, but the result is not usable in the portal. I have to find a workaround for that because the build-in build generates to big js files.

**Comment by me-12** [comment]
@ssi-hu-tasi-norbert Thanks for the help!

Maybe I find some time to add an example in the next days. 

According file size, have a look at this issue: https://github.com/me-12/single-spa-portal-example/issues/34

**Comment by thebeard** [comment]
Thanks for taking the time to comment @me-12. I'm hoping to integrate a few existing Angular projects into a portal too. I started by making sure I can remove the Vue and React apps, to get more familiar with the portal approach.

I then scaffolded ( `ng new` ) a new project into the root directory. I copied over the required logic from `singleSpaEntry.ts` into a file a called `main.platform.ts` into the generated project. I also copied over the `webpack.config.js` file to the folder, added all the `devDependencies` to the new projects `package.json` file and added the two webpack scripts.

At the moment I can run `ng serve` in the scaffolded project ( using it stand-alone ), but running the webpack script ( `npm run watch:portal` ) fails.

I get to the point where the web pack server actually starts to run, but when my scaffolded `app.component.ts` file is reached I get `Module not found: Error: Can't resolve 'css-loader'` errors. I tried to add all the loaders ( sass-loader, postcss-loader, etc. ) to the project, but it seems to be a slippery slope from there and it might just be that I'm missing something small.

Any chance you have some advice here?

**Comment by fireappsAdmin** [comment]
You need to install [css-loader]( https://github.com/webpack-contrib/css-loader ) in your angular project. Somewhere in your code is looking for it. 
Here's my css and sass configs for my webconfig. 
`      {
        test: /\.css$/,
	loaders: ['to-string-loader', 'css-loader', 'postcss-loader', 'resolve-url-loader', ]
	},{
        test: /\.(sass|scss)$/,
	loaders: ['css-to-string-loader', 'css-loader', 'postcss-loader', 'resolve-url-loader',  'sass-loader?sourceMap']
}`
